src/dune/lib_rules.ml

```diff
@@ -207,8 +208,9 @@ let build_stubs lib ~cctx ~dir ~expander ~requires ~dir_contents
   let sctx = Compilation_context.super_context cctx in
   let lib_o_files =
     let foreign_sources =
-      Dir_contents.foreign_sources_of_library dir_contents
-        ~name:(Library.best_name lib)
+      let foreign_sources = Dir_contents.foreign_sources dir_contents in
+      let name = Library.best_name lib in
```

Comment:
Stylistic remark: I don't understand why you prefer `let name = ... in foo ~name`. I think it's both more verbose and to understand. The name `name` is somewhat ambiguous in this context too.

Reply:
In this case, I actually just copy-pasted this particular bit without much thinking. But there are a couple of reasons why I prefer this style even if it's a bit more verbose and clumsy looking.

* It's easy to copy/paste/diff blocks of `let .. in` over big function applications with labled args.

* In dune, the projection paths for values change quite often. So you end up changing the value inside the the `~foo:(..)` and it's a bit less convenient.

* We often forget to share common sub-expressions. It usually doesn't matter, but it can perf a bit and does make code harder to read for me. This style encourages to think about the values that can be re-used.

If we had a refactoring tool to inline `let` bindings automatically (and extract them when doing refactoring), I would prefer the inlined style more.

Reply:
Fair enough. I'll keep these concerns in mind in the future.